firebase-database/src/androidTest/java/com/google/firebase/database/QueryTest.java

```diff
@@ -3401,6 +3406,101 @@ public void onComplete(DatabaseError error, DatabaseReference ref) {
     IntegrationTestHelpers.waitFor(semaphore);
   }
 
+  @Test
+  public void emptyQueryGet() throws DatabaseException, InterruptedException {
+    DatabaseReference node = IntegrationTestHelpers.getRandomNode();
+    DatabaseConfig cfg = IntegrationTestHelpers.newTestConfig();
+    final Semaphore semaphore = new Semaphore(0);
+    node.get()
+        .addOnCompleteListener(
+            new OnCompleteListener<DataSnapshot>() {
+              @Override
+              public void onComplete(@NonNull Task<DataSnapshot> task) {
+                assertTrue(task.isSuccessful());
+                assertNotNull(task.getException());
+                assertEquals(task.getException().getMessage(), "Client offline with empty cache!");
```

Comment:
Nit: Drop the exclamation mark.

But my larger question is why we are offline here. Is this because we kick off going online only after this API call finishes (like we discussed before)?

Reply:
My intent for this test was to check that the proper exception is returned by get() when the client is offline.

Reply:
That is a mistake, this test should never be offline.